django/core/management/commands/shell.py

```diff
@@ -79,6 +62,27 @@ def python(self, options):
                 except Exception:
                     traceback.print_exc()
 
+        # By default, this will set up readline to do tab completion and to read and
+        # write history to the .python_history file, but this can be overridden by
+        # $PYTHONSTARTUP or ~/.pythonrc.py.
+        try:
+            sys.__interactivehook__()
+        except Exception:
+            # Match the behavior of the cpython shell where an error in
+            # sys.__interactivehook__ prints a warning and the exception and continues.
+            print("Failed calling sys.__interactivehook__")
+            traceback.print_exc()
+
+        # Set up tab completion for objects imported by $PYTHONSTARTUP or
+        # ~/.pythonrc.py.
+        try:
+            import readline
+            import rlcompleter
+            readline.set_completer(rlcompleter.Completer(imported_objects).complete)
+        except ImportError:
+            pass
```

Comment:
Hey @inglesp — can you crib for me why we need this block? 🤔 Thanks.

Reply:
It's OK, I see — `imported_objects` is used as the interpreter environment, and we `exec` `.pythonrc` into it. — I think I was looking at the wrong diff. 😜